src/lib/controllers/environment-file/environmentEditor.ts

```diff
@@ -0,0 +1,300 @@
+import inquirer = require('inquirer');
+
+import { get, merge, omit } from 'lodash';
+
+import {
+	Bastion,
+	ConfigStore,
+	DeployTarget,
+	EnvironmentEditorArgs,
+	EnvironmentObject,
+} from '.';
+import {
+	configStoreTypes,
+	deployTargetTypes,
+	loadFromFileSync,
+	writeYamlSync,
+} from '../../tools';
+import {
+	inquirerValidateDirectory,
+	inquirerValidateFQDN,
+	inquirerValidatePath,
+	inquirerValidatePort,
+	inquirerValidateString,
+} from '../../validators';
+
+import { Environment } from './environment';
+
+export class EnvironmentEditor {
+	static async inquireBastion(defaultBastion?: Bastion): Promise<object> {
+		const questions = [
+			{
+				message: 'Please enter your bastion hostname',
+				type: 'input',
+				name: 'bastionHost',
+				default: get(defaultBastion, 'bastionHost'),
+				validate: inquirerValidateFQDN,
+			},
+			{
+				message: 'Please enter your bastion port',
+				type: 'input',
+				name: 'bastionPort',
+				default: get(defaultBastion, 'bastionPort'),
+				validate: inquirerValidatePort,
+			},
+			{
+				message: 'Please enter your bastion username',
+				type: 'input',
+				name: 'bastionUsername',
+				default: get(defaultBastion, 'bastionUsername'),
+				validate: inquirerValidateString,
+			},
+			{
+				message: 'Please enter your bastion key path',
+				type: 'input',
+				name: 'bastionKey',
+				default: get(defaultBastion, 'bastionKey'),
+				validate: inquirerValidatePath,
+			},
+			{
+				message: 'Please enter your bastion key password (if any)',
+				type: 'input',
+				name: 'bastionKeyPassword',
+				default: get(defaultBastion, 'bastionKeyPassword'),
+			},
+		];
+		return inquirer.prompt(questions);
+	}
+
+	verbose: boolean;
```

Comment:
This, `environment` and `configurationPath` are all public by default, is this what you wanted?

Reply:
They were, but they can definitely be `private` and `private readonly` accordingly